runtime/cudaq/qis/execution_manager.h

```diff
@@ -31,6 +31,35 @@ struct QuditInfo {
       : levels(_levels), id(_id) {}
 };
 
+bool __nvqpp__MeasureResultBoolConversion(int);
+
+#ifdef CUDAQ_LIBRARY_MODE
+
+/// @brief In library mode, we model the return type of 
+/// a qubit measurement result via the measure_result type. 
+/// This allows us to keep track of when the result is 
+/// implicitly casted to a bool (likely in the case of 
+/// conditional feedback), and affect the simulation accordingly
+class measure_result {
+private:
+  /// @brief The intrinsic measurement result
+  int result = 0;
```

Comment:
Not sure about `int` here. The result of a measurement is a single bit or a vector of single bits.

Reply:
For qudits it can be an integer.